internal/driver/gomobile/file.go

```diff
@@ -56,7 +62,10 @@ func ShowFileOpenPicker(callback func(fyne.URIReadCloser, error), filter storage
 				callback(nil, nil)
 				return
 			}
-			f, err := fileReaderForURI(storage.NewURI(uri))
+			f, err := fileReaderForURI(&mobileURI{
```

Comment:
Only Android provides this extension, but you're returning it for iOS as well - is that intentional? 
Maybe this init should change - or maybe the code should move to `repository.go` out from the android code?

Reply:
I figured if iOS needs to use it, it can Override Name() like I'm doing, otherwise just does nothing and embedded URI is used.